src/Http/Cookie/Cookie.php

```diff
@@ -0,0 +1,506 @@
+    }
+
+    /**
+     * Sets the expiration date
+     *
+     * @param \DateTimeInterface $dateTime Date time object
+     * @return $this
+     */
+    public function expiresAt(DateTimeInterface $dateTime)
+    {
+        $this->expiresAt = (int)$dateTime->format('U');
+
+        return $this;
+    }
+
+    /**
+     * Checks if a value exists in the cookie data
+     *
+     * @param string $path Path to check
+     * @return bool
+     */
+    public function check($path)
+    {
+        return Hash::check($this->value, $path);
+    }
+
+    /**
+     * Writes data to the cookie
+     *
+     * @param string $path Path to write to
+     * @param mixed $value Value to write
+     * @return $this
+     */
+    public function write($path, $value)
+    {
+        if (!$this->isExpanded) {
+            throw new RuntimeException('The Cookie data has not been expanded');
+        }
+
+        Hash::insert($this->value, $path, $value);
+
+        return $this;
+    }
+
+    /**
+     * Read data from the cookie
+     *
+     * @param string $path Path to read the data from
+     * @return mixed
+     */
+    public function read($path = null)
+    {
+        if (!$this->isExpanded) {
+            throw new RuntimeException('The Cookie data has not been expanded');
+        }
+
+        if ($path === null) {
+            return $this->value;
+        }
+
+        return Hash::get($this->value, $path);
+    }
+
+    /**
+     * Sets the cookies date to a far future so it will virtually never expire
+     *
+     * @return $this
+     */
+    public function willNeverExpire()
+    {
+        $this->expiresAt = Chronos::now()->setDate(2038, 1, 1)->format('U');
+
+        return $this;
+    }
+
+    /**
+     * Deletes the cookie from the browser
+     *
+     * This is done by setting the expiration time to "now"
+     *
+     * @return $this
+     */
+    public function willBeDeleted()
+    {
+        $this->expiresAt = Chronos::now()->format('U');
+
+        return $this;
+    }
+
+    /**
+     * Encrypts the cookie value
+     *
+     * @param string $key Encryption key
+     * @return $this
+     */
+    public function encrypt($key)
+    {
+        $this->encryptionKey = $key;
+        $this->value = $this->_encrypt(
+            $this->value,
+            $this->encryptionCipher,
+            $key
+        );
+
+        return $this;
+    }
+
+    /**
+     * Decrypts the cookie value
+     *
+     * @param string $key Encryption key
+     * @return $this
+     */
+    public function decrypt($key)
+    {
+        $this->encryptionKey = $key;
+        $this->value = $this->_decrypt(
+            $this->value,
+            $this->encryptionCipher,
+            $key
+        );
+
+        return $this;
+    }
+
+    /**
+     * Expands a serialized cookie value
+     *
+     * @return $this
+     */
+    public function expand()
+    {
+        if (!$this->isExpanded) {
+            $this->data = $this->_expand($this->value);
```

Comment:
Nice catch from PHPStan, `$this->data` is not defined nor used anywhere else. Should be `$this->value` here